src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpOntologyDAO.scala

```diff
@@ -1,53 +1,62 @@
   extends OntologyDAO with RestJsonClient with LazyLogging {
 
-  override def search(term: String): Future[Option[List[TermResource]]] = {
-    searchBlocking(term)
+  private val ontologyUri = Uri(FireCloudConfig.Duos.baseOntologyUrl)
+
+  private val (ontologyPort, sslEncryption) = (ontologyUri.authority.port, ontologyUri.scheme) match {
+    case (0, "https") => (443, true)
+    case (0, "http") => (80, false)
+    case (port:Int, "https") => (port, true)
+    case (port:Int, "http") => (port, false)
+    case _ => throw new FireCloudException(s"Could not parse ontologyUri: ${ontologyUri.toString}")
   }
 
-  @deprecated("uses blocking I/O, boo!", "2017-03-15")
-  private def searchBlocking(term: String): Future[Option[List[TermResource]]] = {
-    val targetUri = Uri(ontologySearchUrl).withQuery(("id", term))
-    logger.info(s"HttpOntologyDAO querying ${term} ...")
-    val response = Await.result(unAuthedRequest(Get(targetUri)), Duration(30, "seconds"))
-    logger.info(s"HttpOntologyDAO querying ${term}: ${response.status.defaultMessage}")
-    response.entity.as[List[TermResource]] match {
-      case Right(obj) => Future(Some(obj))
-      case Left(err) =>
-        logger.warn(s"Error while retrieving ontology parents for id '$term': $err")
-        Future(None)
-    }
+  private val ontologyHostSetup = Http.HostConnectorSetup(ontologyUri.authority.host.address, ontologyPort, sslEncryption)
+
+  override def search(term: String): Future[Option[List[TermResource]]] = {
+    searchAsync(term)
   }
 
   private def searchAsync(term: String): Future[Option[List[TermResource]]] = {
-    val targetUri = Uri(ontologySearchUrl).withQuery(("id", term))
-    logger.info(s"HttpOntologyDAO querying ${term} ...")
-    unAuthedRequest(Get(targetUri)) map { response =>
-      logger.info(s"HttpOntologyDAO querying ${term}: ${response.status.defaultMessage}")
-      response.entity.as[List[TermResource]] match {
-        case Right(obj) => Some(obj)
-        case Left(err) =>
-          logger.warn(s"Error while retrieving ontology parents for id '$term': $err")
-          None
+    getHostConnector flatMap { hostConnector =>
+      val targetUri = Uri(ontologySearchUrl).withQuery(("id", term))
+      logger.debug(s"HttpOntologyDAO querying ${term} ...")
+      unAuthedRequest(Get(targetUri), connector = Some(hostConnector)) map { response =>
+        logger.debug(s"HttpOntologyDAO querying ${term}: ${response.status.defaultMessage}")
+        response.entity.as[List[TermResource]] match {
+          case Right(obj) => Some(obj)
+          case Left(err) =>
+            logger.warn(s"Error while retrieving ontology parents for id '$term': $err")
+            None
+        }
       }
     }
   }
 
+  private def getHostConnector: Future[ActorRef] = {
+    implicit val timeout:Timeout = 60.seconds // timeout to get the host connector reference
+    for (Http.HostConnectorInfo(connector, _) <- IO(Http) ? ontologyHostSetup) yield connector
+  }
```

Comment:
The timeout here is the timeout to get the host connector object from the pool. It is NOT the timeout for the request to ontology.